<?xml version="1.0"?>
<robot name="delto_3f" xmlns:xacro="http://www.ros.org/wiki/xacro">

    <xacro:include filename="$(find delto_description)/urdf/delto_gripper_3f.urdf" />

    <link name="base_link" />

    <joint name="base_joint" type="fixed">
        <parent link="base_link"/>
        <child link="delto_base_link"/>
        <origin xyz="0 0 0" rpy="0 0 0"/>
    </joint>

    <ros2_control name="delto_3f_driver" type="system">
        <hardware>
            <plugin>delto_3f_driver/SystemInterface</plugin>
            <param name="delto_ip">$(arg delto_ip)</param>
            <param name="delto_port">$(arg delto_port)</param>
            <param name="p_gain" default="10.0">$(arg p_gain)</param>
            <param name="d_gain" default="1.0">$(arg d_gain)</param>
        </hardware>

        <!-- Finger 1 joints -->
        <joint name="F1M1">
            <command_interface name="effort"/>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
        </joint>
        <joint name="F1M2">
            <command_interface name="effort"/>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
        </joint>
        <joint name="F1M3">
            <command_interface name="effort"/>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
        </joint>
        <joint name="F1M4">
            <command_interface name="effort"/>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
        </joint>

        <!-- Finger 2 joints -->
        <joint name="F2M1">
            <command_interface name="effort"/>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
        </joint>
        <joint name="F2M2">
            <command_interface name="effort"/>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
        </joint>
        <joint name="F2M3">
            <command_interface name="effort"/>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
        </joint>
        <joint name="F2M4">
            <command_interface name="effort"/>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
        </joint>

        <!-- Finger 3 joints -->
        <joint name="F3M1">
            <command_interface name="effort"/>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
        </joint>
        <joint name="F3M2">
            <command_interface name="effort"/>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
        </joint>
        <joint name="F3M3">
            <command_interface name="effort"/>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
        </joint>
        <joint name="F3M4">
            <command_interface name="effort"/>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
        </joint>


    </ros2_control>
</robot>